documentation/snip-images.sh

```diff
@@ -6,6 +6,24 @@
 . $(dirname $0)/../multi-platform-support.sh
 
 FILE=$1
+
+# Read the kafka versions file and create an array of version strings
+declare -a versions
+finished=0
+counter=0
+while [ $finished -lt 1 ] 
+do
+    version=$(yq read "$FILE" "[${counter}].version")
+
+    if [ "$version" = "null" ]
+    then
+        finished=1
+    else
+        versions+=("$version")
+        counter=$((counter + 1))
+    fi 
+done
+
```

Comment:
This is the same as what we've seen before. Can we factor out this stuff into some functions in some bash file which we source at use sites like this. Then it's just a matter of calling the function and all the bash fu is kept in a single place. I don't think there's any benefit from the fact that the `kafka-versions.yaml` is being passed as an argument to this script, so let's just hard code it in the file which gets sourced.

Reply:
After copy and pasting for the third time, I was thinking this would be a better way to do it. I have moved all the kafka-versions parsing functions to `tools/kafka-versions-tools.sh`, unless you think there is a better place for it?

Reply:
SGTM

Reply:
So I am hitting an issue with relative paths to the kafka-versions-tools script. What is the best way to solve this as the make file is called from the root but in some instances you can drop into the docker-images dir etc? Should we set a STRIMZI_ROOT at some point and put absolute paths relative to that?

Reply:
Can you do something using `$(dirname $(realpath $0))` to find the script's path, and then find the file to source relative to that? I.e. `$(dirname $(realpath $0))/kafka-versions-tools` for a top level script or `$(dirname $(realpath $0))/../kafka-versions-tools` for a script in documentation, say?

Reply:
That works a treat!